build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchBuildCompletePlugin.java

```diff
@@ -142,10 +142,16 @@ public void execute(BuildFinishedFlowAction.Parameters parameters) throws FileNo
             uploadFile.getParentFile().mkdirs();
             createBuildArchiveTar(parameters.getFilteredFiles().get(), parameters.getProjectDir().get(), uploadFile);
             if (uploadFile.exists() && "true".equals(System.getenv("BUILDKITE"))) {
-                String uploadFilePath = "build/" + uploadFile.getName();
+                String uploadFilePath = uploadFile.getName();
```

Comment:
Is there a reason we can't just pass the absolute path to `buildkite-agent artifact upload`?

Reply:
I'm actually not sure what happens when you use the absolute paths, and the docs don't seem to mention it. I bet the entire path becomes part of the artifact name.